### PR TITLE
fix: style multiselect input

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
Per @ktmud's findings from https://github.com/apache/incubator-superset/pull/11289 there is some jumpy behavior if the 
placeholder is very wide. It looks like react-select is setting a dynamic width on the input based on the length of the placeholder. Then there's an overflow: hidden to cut down the size to fit the component. Together with the existing flex-box styling, it forced the element to go off the element to the left. The width element is somewhat fighting with the flex-box functionality, so I overwrote it and relied solely on flexbox to determine the width of the elements and let it adjust more gracefully. I'm not 100% familiar with the react-select library, so there may be more tooling hid somewhere that we can utilize. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
select_dropdown (1)

### TEST PLAN
tested visuallly in storybook

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/stylesheets/superset.less [hunk 1]
```diff
@@ -590,3 +590,9 @@ hr {
   background-image: url('../images/icons/error_solid_small_red.svg') !important;
   background-position: -2px center !important;
 }
+
+.Select__value-container--is-multi .Select__input {
+  input {
+    width: 100% !important;
```
Comment (eschutho) [inline superset-frontend/stylesheets/superset.less:596]: It doesn't look like react-select exposes this.

File: superset-frontend/src/components/Select/styles.tsx [hunk 2]
```diff
@@ -156,6 +156,7 @@ export const DEFAULT_STYLES: PartialStylesConfig = {
           border-color: ${borderColor};
           box-shadow: 0 1px 0 rgba(0, 0, 0, 0.06);
         }
+        flex-wrap: nowrap;
```
Comment (eschutho) [inline superset-frontend/src/components/Select/styles.tsx:159]: there are two elements in the container: the wrapper for the inputs and the indicator. We always want these on one row.

File: superset-frontend/src/components/Select/styles.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ export const DEFAULT_STYLES: PartialStylesConfig = {
   clearIndicator: provider => [
     provider,
     css`
-      padding-right: 0;
+      padding: 4px 0 4px 6px;
```
Comment (eschutho) [inline superset-frontend/src/components/Select/styles.tsx:131]: not necessarily related, but I noted when adding and removing the clear indicator in storybook, this moved around. This only happens in storybook, but made it more consistent nonetheless.

**Comment by ktmud** [comment]
Thanks for the fix! Will verify later.

In the meantime, can we disabled browser's native [autocomplete](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#Disabling_autocompletion) as well?

File: superset-frontend/stylesheets/superset.less (hunk 1 above)
Comment (ktmud) [reply superset-frontend/stylesheets/superset.less:596]: I tested locally and it seems to work better without this line. Plus this looks very very hacky... I'm not sure it's worth it. 


The diff that worked for me:

```diff
diff --git a/superset-frontend/src/components/Select/styles.tsx b/superset-frontend/src/components/Select/styles.tsx
index f8c2e8825..ed476ffde 100644
--- a/superset-frontend/src/components/Select/styles.tsx
+++ b/superset-frontend/src/components/Select/styles.tsx
@@ -312,6 +312,9 @@ export const DEFAULT_COMPONENTS: SelectComponentsType = {
       <Input
         {...props}
         placeholder={isMultiWithValue ? placeholder : undefined}
+        // disable Chrome's address autocomplete
+        // @ts-ignore
+        autocomplete="chrome-off"
         css={getStyles('input', props)}
       />
     );
diff --git a/superset-frontend/stylesheets/superset.less b/superset-frontend/stylesheets/superset.less
index adb56ea11..cdd259b6d 100644
--- a/superset-frontend/stylesheets/superset.less
+++ b/superset-frontend/stylesheets/superset.less
@@ -590,9 +590,3 @@ hr {
   background-image: url('../images/icons/error_solid_small_red.svg') !important;
   background-position: -2px center !important;
 }
-
-.Select__value-container--is-multi .Select__input {
-  input {
-    width: 100% !important;
-  }
-}
```

**Comment by eschutho** [comment]
It looks like the onPaste also wraps the Input in another div, so the styles aren't applied for flexbox in that case. I'm poking around inside react-select a bit more to see if I can pass in some styles that get applied to the input or it's wrapper so that it stays within the bounds of the flexbox component. Also, will turn off the autocomplete.. thanks for the reminder!

**Comment by eschutho** [comment]
Ok, I was able to find a way to pass styles into react-select for the input element. The reason I think it needs to be contained to the parent element is for long placeholders. Here's an example:
_DEV__Region_Filter

I updated the OnPasteSelect as mentioned above by removing the wrapper so that we could continue to utilize the flex box styles, and it appears to work, but lmk if there's maybe some functionality that I didn't test. 

Here are videos of three places that I tested. Some of us are looking into automated visual testing tools as well!
region_filter
chart_filter
storybook

**Comment by eschutho** [comment]
quick note: I wasn't able to get autocomplete to show, but I saw that react-select has `autocomplete="off"` which doesn't work for chrome, so I'm fairly certain that what you had should do the trick.

**Comment by eschutho** [comment]
Tests are taking a long time to run. Restarting.

**Review by ktmud — APPROVED** [review]
This looks great! Thanks for the storybook test cases!

**Comment by eschutho** [comment]
Thanks @ktmud! Feel free to merge when ready.

**Comment by nytai** [comment]
Something in this PR caused an issue with our listview filters width

before this change:
Screen Shot 2020-11-25 at 2 19 23 PM
After this change:
Screen Shot 2020-11-25 at 2 22 50 PM

**Pull request merged**